drivers/dma/dma_stm32f4x.c

```diff
@@ -341,19 +341,23 @@ static int dma_stm32_config_devcpy(struct device *dev, u32_t id,
 static int dma_stm32_config_memcpy(struct device *dev, u32_t id)
 {
 	struct dma_stm32_device *ddata = dev->driver_data;
-	struct dma_stm32_stream_reg *regs = &ddata->stream[id].regs;
+	if (ddata->mem2mem) {
+		struct dma_stm32_stream_reg *regs = &ddata->stream[id].regs;
 
-	regs->scr = DMA_STM32_SCR_DIR(DMA_STM32_MEM_TO_MEM) |
-		DMA_STM32_SCR_MINC |		/* Memory increment mode */
-		DMA_STM32_SCR_PINC |		/* Peripheral increment mode */
-		DMA_STM32_SCR_TCIE |		/* Transfer comp IRQ enable */
-		DMA_STM32_SCR_TEIE;		/* Transfer error IRQ enable */
+		regs->scr = DMA_STM32_SCR_DIR(DMA_STM32_MEM_TO_MEM) |
+			DMA_STM32_SCR_MINC | DMA_STM32_SCR_PINC | DMA_STM32_SCR_TCIE |
+			DMA_STM32_SCR_TEIE;
 
-	regs->sfcr = DMA_STM32_SFCR_DMDIS |	/* Direct mode disable */
-		DMA_STM32_SFCR_FTH(DMA_STM32_FIFO_THRESHOLD_FULL) |
-		DMA_STM32_SFCR_FEIE;		/* FIFI error IRQ enable */
+		regs->sfcr = DMA_STM32_SFCR_DMDIS |	/* Direct mode disable */
+			DMA_STM32_SFCR_FTH(DMA_STM32_FIFO_THRESHOLD_FULL) |
+			DMA_STM32_SFCR_FEIE;		/* FIFI error IRQ enable */
 
-	return 0;
+		return 0;
+	} else {
+		SYS_LOG_ERR("DMA error: Memcopy not supported for device %s",
+					dev->config->name);
+		return -EINVAL;
```

Comment:
Some computation on registers is done in dma_stm32_config even if current function returns -EINVAL:
```
regs->sndtr = config->head_block->block_size;
```
Can you take care of that?